infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -375,12 +347,17 @@ public void startSnapshotSyncApply() {
     }
 
     /**
-     * Clear local updated streams for which no data was transferred from active
-     * (hence were not cleared on the replication/transfer path)
+     * Clear streams which :
+     * 1.has evidenced data from active
+     * 2.has local updates and no data transferred from active
      *
      * Note: streams could be locally written while this node had no assigned role.
      */
     public void clearLocalStreams() {
+        Set<UUID> streamsToClear = new HashSet<>();
+        // add the streams collected during the transfer phase (streams that had incoming data from active)
+        streamsToClear.addAll(replicatedStreamIds);
+
```

Comment:
Just a question, I'm concerning this might be a problem after the dynamic stream list change.

So basically by clearing streams with local writes, I thought we want to sync the state of STANDBY cluster with ACTIVE cluster. But by iterating over the streams to replicate provided by static config, we are somehow "overkilling" some steams. i.e. streams have local writes in STANDBY, included in that list, but not even opened/created at ACTIVE cluster (let's call it `stream set A`).

However, after the dynamic streams change, we cannot get this `stream set A` and cannot clear them as a result. The best we can do in clearing step might be clearing those already opened in ACTIVE (such that registered to Registry Table) but no data transferred.

Could this be a problem if that set of streams are not cleared in this step?

Reply:
yes, we need to clear stream that have local writes but not opened by active..otherwise we will keep passing the pollution (and we have no way of differentiating valid writes and pollution)